### PR TITLE
OOP school library: unit tests

### PR DESCRIPTION
**Pull request by ShiroYaksha90**
## Hi 👋 , in this pull request we:
- Add unit tests for all the methods in all the classes:
  - `Book`
  - `Person`
  - `Student`
  - `Teacher`
  - `Classroom`
  - `Rental`
  - all classes that we created while implementing the Decorator design pattern

- [ ] To run the test consider the following
  - run `bundle install` to add all the dependencies used
  - run `bundle exec rspec .` to excute all the tests

### CODE REVIEW

File: spec/person_spec.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+      expect(@person.age).to eql 15
+    end
+  end
+
+  describe '#name' do
+    it 'Should return Person name' do
+      expect(@person.name).to eql 'Joe'
+    end
+  end
+end
```
Comment (cynthiainga) [inline spec/person_spec.rb]: - [ ] Good job so far, but you need to add some tests and you are good to go 🙃 😁:
  - The context of when creating a new instance of the Person class and the constructor method sets a default value for the optional name parameter if it is not provided 😁 
  - The context of when adding new rentals
    -  The scenario of the add_rental method returns a rental and adds it to the book
    - The scenario of the constructor method returns a Person object with provided age, name, and empty rentals
  - The context when checking permissions:
    - The scenario of the `can_use_services?` method returns false if under age and no parent permission
    - The scenario of the `can_use_services?` method returns true if over age
    - The scenario of the `can_use_services?` method returns true if has parent permission

File: spec/student_spec.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  describe '#name' do
+    it 'Should return student name' do
+      expect(@student.name).to eql 'Rose'
+    end
+  end
+  describe '#parent permission' do
+    it 'Should return parent permission' do
+      expect(@student.parent_permission).to be false
+    end
+  end
```
Comment (cynthiainga) [inline spec/student_spec.rb:26]: - [ ] It would be better to test the `play_hooky` method as well, kindly add it 😁

**Review by cynthiainga — CHANGES_REQUESTED** [review]
Hi @ShiroYaksha90 & @KUBAHO3, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights

- [ ] Good commit history ✔️ 
- [ ] Correct Gitflow ✔️ 
- [ ] All implemented tests are passing so far ✔️ 

## Required Changes ♻️

- [ ] Please kindly check the comments under the review 👇🏾 👇🏾 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._


Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/student_spec.rb (hunk 2 above)
Comment (ShiroYaksha90) [reply spec/student_spec.rb:26]: @cynthiainga Thank you for your insightful review 🌹 
I implemented the required changes 👍

**Review by danmainah — COMMENTED** [review]
Hi @ShiroYaksha90 and @KUBAHO3 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 

## Highlights
✔️  Implemented changes requested to ensure all tests scenarios are covered.
✔️  Esnured all tests are passing.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**